packages/rgbpp/src/rgbpp/utils/transaction.ts

```diff
@@ -22,25 +17,20 @@ export async function sendRgbppTxGroups(props: {
 }): Promise<SentRgbppTxGroup[]> {
   const results: SentRgbppTxGroup[] = [];
   for (const group of props.txGroups) {
-    const result: SentRgbppTxGroup = {
-      ckb: {},
-      btc: {},
-    };
-    try {
-      const sent = await props.btcService.sendBtcTransaction(group.btcTxHex);
-      result.btc.txId = sent.txid;
-    } catch (e) {
-      result.btc.error = e;
-    }
     try {
+      const { txid } = await props.btcService.sendBtcTransaction(group.btcTxHex);
       await props.btcService.sendRgbppCkbTransaction({
-        btc_txid: result.btc.txId!,
+        btc_txid: txid,
         ckb_virtual_result: group.ckbVirtualTxResult,
       });
+      results.push({ btcTxId: txid });
     } catch (e) {
-      result.ckb.error = e;
+      if (e instanceof BtcAssetsApiError) {
+        results.push({ error: e.message });
+      } else {
+        results.push({ error: 'Sending the RGB++ group transactions failed' });
```

Comment:
Maybe add the `e` to the `error` when it's not BtcAssetsApiError?

Reply:
The try-catch block only includes btcService RPC, so I don't think another error will occur.

Reply:
You reminded me that the BtcAssetsApi request could throw a `TypeError` when the `fetch()` fails. I'm not sure if we should wrap the `fetch()` in a try-catch block. The current implementation only catches errors during response parsing. See: [GitHub link](https://github.com/ckb-cell/rgbpp-sdk/blob/develop/packages/service/src/service/base.ts#L38-L57).

```
TypeError: fetch failed
    at node:internal/deps/undici/undici:13737:13
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
    at BtcAssetsApi.request (/Users/root/Development/projects/rgbpp-sdk/packages/service/src/service/base.ts:38:17)
```

Reply:
I think the above error message of the `fetch` function is not helpful for developers

Reply:
https://github.com/ckb-cell/rgbpp-sdk/pull/283/commits/1c11426f6c1ca6ab196929b0e6641ccfdbdc4c20